packages/hawtio/src/plugins/camel/CamelPreferences.tsx

```diff
@@ -0,0 +1,111 @@
+import { CardBody, Checkbox, Form, FormGroup, FormSection, TextInput } from '@patternfly/react-core'
+import React, { useState } from 'react'
+import { camelPreferencesService, ICamelPreferences } from './camel-preferences-service'
+
+export const CamelPreferences: React.FunctionComponent = () => (
+  <CardBody>
+    <Form isHorizontal>
+      <CamelPreferencesForm />
+    </Form>
+  </CardBody>
+)
+
+const CamelPreferencesForm: React.FunctionComponent = () => {
+  const [camelPreferences, setCamelPreferences] = useState(camelPreferencesService.loadCamelPreferences())
+
+  const updatePreferences = (value: boolean | number, key: keyof ICamelPreferences): void => {
+    const updatedPreferences = { ...camelPreferences, ...{ [key]: value } }
+
+    camelPreferencesService.saveCamelPreferences(updatedPreferences)
+    setCamelPreferences(updatedPreferences)
+  }
+
+  const updateNumberValueFor = (key: keyof ICamelPreferences): ((value: string) => void) => {
+    //Returning an arrow function to reduce boilerplate
+    return (value: string) => {
+      const intValue = parseInt(value)
+
+      if (!intValue) return
+
+      updatePreferences(intValue, key)
+    }
+  }
+
+  const updateCheckboxValueFor = (key: keyof ICamelPreferences): ((value: boolean, _: React.FormEvent) => void) => {
+    //Utility function generator to reduce boilerplate
+    return (value: boolean, _: React.FormEvent) => {
+      updatePreferences(value, key)
+    }
+  }
+
+  return (
+    <FormSection>
+      <FormGroup label='Hide option documentation' fieldId='camel-form-hide-option-documentation'>
+        <Checkbox
+          id='camel-form-hide-option-documentation'
+          isChecked={camelPreferences.isHideOptionDocumentation}
+          onChange={updateCheckboxValueFor('isHideOptionDocumentation')}
```

Comment:
They keyof operator makes these "magic strings" be statically typed. If you were to put something like 'a' or 'notAKeyHahahahahaha' Ts will throw a compiler error so they are not really magic strings